lib/heap_sort.rb

```diff
@@ -1,8 +1,21 @@
-
+require_relative "min_heap"
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
-end
\ No newline at end of file
+# Time Complexity:  O(n)
+# Space Complexity: 0(n)
+
+def heapsort(list)
```

Comment:
Since you're adding n elements to a heap (which takes log n time) and them removing them the time complexity is O(n log n).